node/src/components/linear_chain/pending_signatures.rs

```diff
@@ -68,10 +62,10 @@ impl PendingSignatures {
                 %block_hash, %public_key,
                 "received too many finality signatures for unknown blocks"
             );
-            return;
+            return false;
         }
         // Add the pending signature.
-        let _ = sigs.insert(block_hash, signature);
+        sigs.insert(block_hash, signature).is_some()
```

Comment:
```suggestion
        sigs.insert(block_hash, signature).is_none()
```
Would be good to have a unit test for this. (Or for something higher level, that also tests this.)

Reply:
Actually, I think this should always be true anyway. This `add` method should return `true` if the value was added and it doesn't matter whether there was entry under `block_hash` previously or not.

Reply:
Fixed in [`50f753a` (#1268)](https://github.com/CasperLabs/casper-node/pull/1268/commits/50f753a8773becc78c914139bb5035ba830f7631)